distributed/shuffle/shuffle_extension.py

```diff
@@ -302,7 +312,8 @@ async def _barrier(self, shuffle_id: ShuffleId) -> None:
         # Note that this will call `shuffle_inputs_done` on our own worker as well
         shuffle = await self._get_shuffle(shuffle_id)
         out = await self.worker.scheduler.broadcast(
-            msg={"op": "shuffle_inputs_done", "shuffle_id": shuffle_id}
+            msg={"op": "shuffle_inputs_done", "shuffle_id": shuffle_id},
+            workers=set(shuffle.worker_for),
```

Comment:
If we're stricter with input validation, this must only broadcast to participating workers. This is sane in general and should not impact any functionality